### ISSUE TITLE
Representing Percents

### ISSUE THREAD

**Issue by stevenchalem** [semanticarts/gist#785]
1) How should percentages be stored when users want to see hundredths vs fractions (75 vs 0.75)
2) Should percentage be considered a unit of measure?
3) Should percentage be considered multiple units of measure depending on underlying units (e.g. number of correct test questions out of total test questions vs. body fat as a percentage of total weight vs. interest rate)

**Comment by stevenchalem** [comment]
@marksem I think this was your issue. Please correct this ticket if I've got it wrong.

**Comment by rjyounes** [comment]
I believe this could involve either modifications to gist or just a modeling pattern, depending on what we decide.

**Comment by marksem** [comment]
Jotting down some quick thoughts:

A percentage seems like a magnitude, e.g. "% of registered voter turnout".  It seems like a magnitude, like "voter count" would be a Magnitude (Count subclass), with unit _each.  The UoM tells us how to interpret the number.  Percent seems like a UoM, and the magnitude would be what the percentage is regarding.  A "Percent Magnitude" would be a Magnitude whose UoM is _percent. 

So perhaps as gist currently says, "Percentage" is a magnitude, and _percent would be the UoM (an addition to gist).

The range of a percent seems it must be 0.0-100.0, since "Percentages are essentially fractions where the denominator is 100. To show that a number is a percent, we use the percent symbol (%) beside the number. For example, if you got 75 questions right out of 100 on a test (75/100), you would have scored 75%".  Gist says 50% can be represented as 50 or 0.5, but I think the latter is incorrect.

**Comment by uscholdm** [comment]
> The range of a percent seems it must be 0.0-100.0, since "Percentages are essentially fractions where the denominator is 100.

Here is a counterexample. If Johnny's salary went up from $50k to $150k after a job switch,, then:
- new salary is 200% more than the prior salary (i.e. a 200% increase)
- new salary is 300% of the prior salary (i.e. it tripled)

I don't think a negative percent makes any sense, however. But I'm not 100% sure (so to speak).

**Comment by rjyounes** [comment]
To be a stickler about words and names: we should be using "percent" rather than "percentage" across the board. The consensus from a number of source indicates that the former is used with a specific number, while the latter is used in a more general sense, possibly with adjectives like "large." This accords with my sense of current usage:

15%, not 15% percentage
A small percentage, not a small percent

**Comment by uscholdm** [comment]
In English, that's true, so maybe all along the term Percentage was never ideal. It may not be strictly accurate, but I don't think it's that bad, nor is it confusing. Its sounds fine to me to say that 
- a `gist:Percentage` is a number expressed as a percent.  

So if you express 73 as a percent, it translates to .73, which Excel does a nice job of handling.

I have used `gist:_percent` as an instance of `gist:UnitOfMeasure` whose base unit is `gist:_each` with a conversion factor of .01

**Comment by rjyounes** [comment]
> I have used gist:_percent as an instance of gist:UnitOfMeasure whose base unit is gist:_each with a conversion factor of .01.

I like this.

**Comment by uscholdm** [comment]
Its fine as far as it goes. What it does not address is a way to say the meaning of the numerator and denominator is. Phil has an interesting idea about that using Aspects. I'm not sure how well it would fit into what we are doing now. Hopefully not too disruptive.

It would be possible, to have subclasses of Percentage for each kind of Percentage.  E.g. UnemploymentRate. Minimally, there would be a skos definition indicating meaning. Better would be to represent the meaning formally, somehow which might enable knowing whether comparing two percentages makes sense.

**Comment by rjyounes** [comment]
Having subclasses like UnemploymentRate seems to me to be flawed in the same way that having a PayRate subclass of Magnitude is. $100/hr is a MonetaryPerDuration magnitude; pay rate is an _aspect_ - i.e., something that is  expressed with such a magnitude. $100/hr is the same object whether it's a pay rate or a billing rate. UnemploymentRate strikes me as the same thing; 5% is 5% whether it's an unemployment rate or a medical school admission rate. It would be as if we were to have classes for Length, Width, and Height  etc. for every _aspect_ that is expressed with an Extent.

**Comment by uscholdm** [comment]
Yes, I think I agree with those points. So, unemployment rate would be an instance of Aspect (or say Characteristic as we have used for many years) that may be applied to some geo entity that took on values that were instances of Percentage.  I think Phil's idea is to use numerator and denominator on such characteristics themselves, as opposed to on the unit which we normally do. That has promise.

**Comment by DanCarey404** [comment]
Based on Phil's notes above and other conversations, I believe we want to consider the following:

**1) Addition of a gist:_percent unit of measure:**
###  https://ontologies.semanticarts.com/gist/_percent
gist:_percent rdf:type owl:NamedIndividual ,
                       gist:RatioUnit ;
              gist:hasBaseUnit gist:_each ;
              gist:conversionFactor "0.01"^^xsd:double ;
              gist:unitSymbol "%"^^xsd:string ;
              rdfs:isDefinedBy <https://ontologies.semanticarts.com/o/gistCore> ;
              skos:definition "The unit designating a quantity that is the ratio of two quantities of the same kind and which is multiplied by 100." ;
              skos:prefLabel "%"^^xsd:string .



**2) Updating the gist:Percentage class** (or alternatively, deleting the Percentage class):
###  https://ontologies.semanticarts.com/gist/Percentage
gist:Percentage rdf:type owl:Class ;
                rdfs:subClassOf gist:RatioMagnitude ,
                                [ rdf:type owl:Restriction ;
                                  owl:onProperty gist:hasUnitOfMeasure ;
                                  owl:hasValue gist:_percent
                                ] ;
                rdfs:isDefinedBy <https://ontologies.semanticarts.com/o/gistCore> ;
                skos:definition "A ratio where the numerator and denominator are of the same unit of measure and the resulting quantity is multiplied by 100, expressed as 'per cent', i.e., 'parts per hundred'."^^xsd:string ;
                skos:prefLabel "Percentage"^^xsd:string ;
                skos:scopeNote "Members of this class are not pure Ratio Magnitudes because they are multiplied by 100 and given the gist:_percent unit of measure, i.e., '73%' not '0.73'."^^xsd:string .

**Comment by uscholdm** [comment]
This is close, but

> ```
> skos:scopeNote "Members of this class are not pure Ratio Magnitudes because they are multiplied by 100 and given the gist:_percent unit of measure, i.e., '73%' not '0.73'."^^xsd:string .
> ```

50km per hour is a ratio magnitude that happens to have a conversion factor or 1000 - that would mean it also is not a pure ration magnitude, but  I don't think there is any utility to a notion of a pure ratio magnitude. 

Also, it might be worth saying that Percentage is a subclass of gist:Count - although it is inferable,

**Comment by uscholdm** [comment]
> skos:definition "A ratio where the numerator and denominator are of the same unit of measure and the resulting quantity is multiplied by 100, expressed as 'per cent', i.e., 'parts per hundred'."^^xsd:string ;

I would rephrase this as:

> skos:definition "A ratio where the numerator and denominator are of the same unit of measure and the resulting dimensionless quantity is a number multiplied by 100, expressed as 'per cent', i.e., 'parts per hundred'."^^xsd:string ;

**Comment by uscholdm** [comment]
> gist:_percent rdf:type owl:NamedIndividual ,

No need to say a named individual.

**Comment by rjyounes** [comment]
From Michael at a previous client:

```
pleox:_UnitOfMeasure_percent
               a gist:UnitOfMeasure ;
               skos:definition "Part per hundred"^^xsd:string ;
               skos:prefLabel "percent"^^xsd:string ;
               skos:scopeNote "This is a real percent with a conversion of .01. Should not be used with gist:Percentage which ignores the number 100. It is a simple ratio of two Magnitudes with the same unit."^^xsd:string ;
               gist:convertToBase ".01"^^xsd:double ;
               gist:hasBaseUnit gist:_each ;
               .
 
```

**Comment by rjyounes** [comment]
```
gist:_percent 
               gist:RatioUnit ;
              gist:hasBaseUnit gist:_each ;
              gist:conversionFactor "0.01"^^xsd:double ;
              gist:unitSymbol "%"^^xsd:string ;
              skos:definition "The unit designating a quantity that is the ratio of two quantities of the same kind and which is multiplied by 100." ;
              skos:prefLabel "percent"^^xsd:string .
```

**Comment by uscholdm** [comment]
See also: #616
